warehouse/accounts/__init__.py

```diff
@@ -206,19 +76,18 @@ def includeme(config):
         breached_pw_class.create_service, IPasswordBreachedService
     )
 
-    # Register our authentication and authorization policies
-    config.set_authentication_policy(
-        MultiAuthenticationPolicy(
-            [
-                SessionAuthenticationPolicy(callback=_session_authenticate),
-                BasicAuthAuthenticationPolicy(check=_basic_auth_check),
-                MacaroonAuthenticationPolicy(callback=_macaroon_authenticate),
-            ]
-        )
+    # Register our security policies (AuthN + AuthZ)
+    authz_policy = TwoFactorAuthorizationPolicy(
+        policy=MacaroonAuthorizationPolicy(policy=ACLAuthorizationPolicy())
```

Comment:
If the goal is to completely switch over to 2.0-style policies, we should probably eventually fold the big callback tower into a single AuthZ policy "helper", since that's the pattern that Pyramid itself uses for e.g. basic and session-based auth 🙂